File: t/CS/JpxDemo/UI/AppendCommon.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using Leadtools.Demos;

namespace JPXDemo
{
   public partial class AppendCommon : Form
   {
      private string _dialogName;
      private bool _showFilterType;
      private AppendCommonStructure _appendCommonData;
      private List<string> _filterTypes;
      private byte[] pGZIP = {0xEC, 0x34, 0x0B, 0x04, 0x74, 0xC5, 0x11, 0xD4, 0xA7, 0x29, 0x87, 0x9E, 0xA3, 0x54, 0x8F, 0x0E };
      private byte[] pDES  = {0xEC, 0x34, 0x0B, 0x04, 0x74, 0xC5, 0x11, 0xD4, 0xA7, 0x29, 0x87, 0x9E, 0xA3, 0x54, 0x8F, 0x0F };

      public List<string> FilterTypes
      {
         get
         {
            return _filterTypes;
         }
         set
         {
            _filterTypes = value;
         }
      }

      public AppendCommonStructure AppendCommonData
      {
         get
         {
            return _appendCommonData;
         }
         set
         {
            _appendCommonData = value;
         }
      }

      public string DialogName
      {
         get
         {
            return _dialogName;
         }
         set
         {
            _dialogName = value;
         }
      }

      public bool ShowFilterType
      {
         get
         {
            return _showFilterType;
         }
         set
         {
            _showFilterType = value;
         }
      }

      public string Jpeg2000FileName
      {
         get
         {
            return _txtJPEG2000.Text;
         }
         set
         {
            _txtJPEG2000.Text = value;
         }
      }

      public string SecondLabel
      {
         get
         {
            return _lblSecond.Text;
         }
         set
         {
            _lblSecond.Text = value;
         }
      }

      public AppendCommon()
      {
         InitializeComponent();

         ShowFilterType = false;
         DialogName = "Append Common";
         FilterTypes = null;

         AppendCommonData = new AppendCommonStructure();

         InitClass();
      }

      public AppendCommon(string dialogName, bool showFilterType, List<String> filterItems)
      {
         InitializeComponent();

         ShowFilterType = showFilterType;
         DialogName = dialogName;

         FilterTypes = filterItems;

         InitClass();
      }

      private void InitClass()
      {
         this.Text = "Append " + DialogName + " Box Dialog";

         SecondLabel = DialogName + " Box Data File:";

         ShowFilterTypeControls(ShowFilterType);
      }

      private void ShowFilterTypeControls(object showFilterType)
      {
         if (ShowFilterType)
         {
            for (int index = 0; index < FilterTypes.Count; index++)
            {
               _cbFilterType.Items.Add(FilterTypes[index]);
            }
            _cbFilterType.SelectedIndex = 0;
            return;
         }

         _lblFilterType.Visible = false;
         _cbFilterType.Visible = false;

         _grpData.Height -= _cbFilterType.Height;
         _lblSecond.Location = new Point(_lblSecond.Location.X, _lblSecond.Location.Y - _cbFilterType.Height);
         _txtSecond.Location = new Point(_txtSecond.Location.X, _txtSecond.Location.Y - _cbFilterType.Height);
         _btnSecondBrowse.Location = new Point(_btnSecondBrowse.Location.X, _btnSecondBrowse.Location.Y - _cbFilterType.Height);
         _btnOk.Location = new Point(_btnOk.Location.X, _btnOk.Location.Y - _cbFilterType.Height);
         _btnCancel.Location = new Point(_btnCancel.Location.X, _btnCancel.Location.Y - _cbFilterType.Height);
         this.Height -= _cbFilterType.Height;
      }

      private void _btnFirstBrowse_Click(object sender, EventArgs e)
      {
         OpenFileDialog ofd = new OpenFileDialog();

         ofd.Title = "Open a File";
         ofd.Filter = "JPX Files(*.jpx)|*.jpx|JP2 Files(*.jp2)|*.jp2";

         if (ofd.ShowDialog() == DialogResult.OK)
         {
            _txtJPEG2000.Text = ofd.FileName;
         }
      }

      private void _btnSecondBrowse_Click(object sender, EventArgs e)
      {
         OpenFileDialog ofd = new OpenFileDialog();

         ofd.Title = "Open a File";
         ofd.Filter = "All Files (*.*)|*.*";

         if (ofd.ShowDialog() == DialogResult.OK)
         {
            _txtSecond.Text = ofd.FileName;
         }
      }

      private void _btnOk_Click(object sender, EventArgs e)
      {
         if (_txtJPEG2000.Text == "")
         {
            Messager.ShowError(this, "Please set the Jpeg2000 file");
            return;
         }

         if (_txtSecond.Text == "")
         {
            Messager.ShowError(this, "Please set the Data file");
            return;
         }

         using (WaitCursor wait = new WaitCursor())
         {
            try
            {
               AppendCommonStructure temp = new AppendCommonStructure();
               if (ShowFilterType)
               {
                  switch (_cbFilterType.SelectedIndex)
                  {
                     case 0: temp.type.Id = pGZIP; break;
                     case 1: temp.type.Id = pDES; break;
                  }
               }

               temp.data = File.ReadAllBytes(_txtSecond.Text);
               AppendCommonData = temp;
               this.DialogResult = DialogResult.OK;
               this.Close();
            }
            catch (Exception ex)
            {
               Messager.ShowError(this, ex);
            }
         }
      }
   }
}

File: t/VB/JPXDemo/UI/AppendCommon.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System.IO
Imports System.Collections.Generic

Public Class AppendCommon
   Private _dialogName As String
   Private _showFilterType As Boolean
   Private _appendCommonData As AppendCommonStructure
   Private _filterTypes As List(Of String)
   Private pGZIP() As Byte = { _
                              Convert.ToByte(Val("0xEC")), _
                              Convert.ToByte(Val("0x34")), _
                              Convert.ToByte(Val("0x0B")), _
                              Convert.ToByte(Val("0x04")), _
                              Convert.ToByte(Val("0x74")), _
                              Convert.ToByte(Val("0xC5")), _
                              Convert.ToByte(Val("0x11")), _
                              Convert.ToByte(Val("0xD4")), _
                              Convert.ToByte(Val("0xA7")), _
                              Convert.ToByte(Val("0x29")), _
                              Convert.ToByte(Val("0x87")), _
                              Convert.ToByte(Val("0x9E")), _
                              Convert.ToByte(Val("0xA3")), _
                              Convert.ToByte(Val("0x54")), _
                              Convert.ToByte(Val("0x8F")), _
                              Convert.ToByte(Val("0x0E "))}

   Private pDES() As Byte = { _
                              Convert.ToByte(Val("0xEC")), _
                              Convert.ToByte(Val("0x34")), _
                              Convert.ToByte(Val("0x0B")), _
                              Convert.ToByte(Val("0x04")), _
                              Convert.ToByte(Val("0x74")), _
                              Convert.ToByte(Val("0xC5")), _
                              Convert.ToByte(Val("0x11")), _
                              Convert.ToByte(Val("0xD4")), _
                              Convert.ToByte(Val("0xA7")), _
                              Convert.ToByte(Val("0x29")), _
                              Convert.ToByte(Val("0x87")), _
                              Convert.ToByte(Val("0x9E")), _
                              Convert.ToByte(Val("0xA3")), _
                              Convert.ToByte(Val("0x54")), _
                              Convert.ToByte(Val("0x8F")), _
                              Convert.ToByte(Val("0x0F "))}

   Public Property FilterTypes() As List(Of String)
      Get
         Return _filterTypes
      End Get
      Set(ByVal value As List(Of String))
         _filterTypes = value
      End Set
   End Property

   Public Property AppendCommonData() As AppendCommonStructure
      Get
         Return _appendCommonData
      End Get
      Set(ByVal value As AppendCommonStructure)
         _appendCommonData = value
      End Set
   End Property

   Public Property DialogName() As String
      Get
         Return _dialogName
      End Get
      Set(ByVal value As String)
         _dialogName = value
      End Set
   End Property

   Public Property ShowFilterType() As Boolean
      Get
         Return _showFilterType
      End Get
      Set(ByVal value As Boolean)
         _showFilterType = value
      End Set
   End Property

   Public Property Jpeg2000FileName() As String
      Get
         Return _txtJPEG2000.Text
      End Get
      Set(ByVal value As String)
         _txtJPEG2000.Text = value
      End Set
   End Property

   Public Property SecondLabel() As String
      Get
         Return _lblSecond.Text
      End Get
      Set(ByVal value As String)
         _lblSecond.Text = value
      End Set
   End Property

   Public Sub New()
      InitializeComponent()

      ShowFilterType = False
      DialogName = "Append Common"
      FilterTypes = Nothing

      AppendCommonData = New AppendCommonStructure()

      InitClass()
   End Sub

   Public Sub New(ByVal dialogName As String, ByVal bShowFilterType As Boolean, ByVal filterItems As List(Of String))
      InitializeComponent()

      ShowFilterType = bShowFilterType
      _dialogName = dialogName

      FilterTypes = filterItems

      InitClass()
   End Sub

   Private Sub InitClass()
      Me.Text = "Append " + DialogName + " Box Dialog"

      SecondLabel = DialogName + " Box Data File:"

      ShowFilterTypeControls(ShowFilterType)
   End Sub

   Private Sub ShowFilterTypeControls(ByVal showFilterType As Boolean)
      If (showFilterType) Then
         For index As Integer = 0 To FilterTypes.Count - 1
            _cbFilterType.Items.Add(FilterTypes(index))
         Next index
         _cbFilterType.SelectedIndex = 0
         Return
      End If

      _lblFilterType.Visible = False
      _cbFilterType.Visible = False

      _grpData.Height -= _cbFilterType.Height
      _lblSecond.Location = New Point(_lblSecond.Location.X, _lblSecond.Location.Y - _cbFilterType.Height)
      _txtSecond.Location = New Point(_txtSecond.Location.X, _txtSecond.Location.Y - _cbFilterType.Height)
      _btnSecondBrowse.Location = New Point(_btnSecondBrowse.Location.X, _btnSecondBrowse.Location.Y - _cbFilterType.Height)
      _btnOk.Location = New Point(_btnOk.Location.X, _btnOk.Location.Y - _cbFilterType.Height)
      _btnCancel.Location = New Point(_btnCancel.Location.X, _btnCancel.Location.Y - _cbFilterType.Height)
      Me.Height -= _cbFilterType.Height
   End Sub

   Private Sub _btnJPEG2000Browse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _btnJPEG2000Browse.Click
      Dim ofd As New OpenFileDialog()

      ofd.Title = "Open a File"
      ofd.Filter = "JPX Files(*.jpx)|*.jpx|JP2 Files(*.jp2)|*.jp2"

      If (ofd.ShowDialog() = System.Windows.Forms.DialogResult.OK) Then
         _txtJPEG2000.Text = ofd.FileName
      End If
   End Sub

   Private Sub _btnSecondBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _btnSecondBrowse.Click
      Dim ofd As New OpenFileDialog()

      ofd.Title = "Open a File"
      ofd.Filter = "All Files (*.*)|*.*"

      If (ofd.ShowDialog() = System.Windows.Forms.DialogResult.OK) Then
         _txtSecond.Text = ofd.FileName
      End If
   End Sub

   Private Sub _btnOk_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _btnOk.Click
      If (_txtJPEG2000.Text = "") Then
         Messager.ShowError(Me, "Please set the Jpeg2000 file")
         Return
      End If

      If (_txtSecond.Text = "") Then
         Messager.ShowError(Me, "Please set the Data file")
         Return
      End If


      Dim wait As New WaitCursor()
      Try
         Dim temp As New AppendCommonStructure()
         If (ShowFilterType) Then
            Select Case (_cbFilterType.SelectedIndex)
               Case 0
                  temp.type.Id = pGZIP
               Case 1
                  temp.type.Id = pDES
            End Select
         End If

         temp.data = File.ReadAllBytes(_txtSecond.Text)
         AppendCommonData = temp
         Me.DialogResult = System.Windows.Forms.DialogResult.OK
         Me.Close()
      Catch ex As Exception
         Messager.ShowError(Me, ex)
      Finally
         wait.Dispose()
      End Try
   End Sub
End Class
